pkg/prometheus/operator.go

```diff
@@ -89,43 +90,12 @@ func New(c Config, logger log.Logger) (*Operator, error) {
 	if err != nil {
 		return nil, err
 	}
-	return &Operator{
+	c := &Operator{
 		kclient: client,
 		pclient: promclient,
 		logger:  logger,
-		queue:   newQueue(200),
+		queue:   queue.New(),
 		host:    cfg.Host,
-	}, nil
-}
-
-// Run the controller.
-func (c *Operator) Run(stopc <-chan struct{}) error {
-	defer c.queue.close()
-	go c.worker()
-
-	errChan := make(chan error)
```

Comment:
Did you mean to overwrite this? I've got to make a small fix to WaitForTPR so can add back in in that PR if you didn't.